sources/models/src/modeled_types/kubernetes.rs

```diff
@@ -994,3 +995,121 @@ mod test_topology_manager_policy {
         }
     }
 }
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// KubernetesClusterDNSIp represents the --cluster-dns settings for kubelet.
+///
+/// The default is determined via information from IMDS in AWS.
+/// This model allows the value to be either a list of IPs, or a single IP string
+/// for backwards compatibility.
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum KubernetesClusterDNSIp {
+    Scalar(IpAddr),
+    Vector(Vec<IpAddr>),
+}
+
+impl KubernetesClusterDNSIp {
+    pub fn iter<'a>(&'a self) -> Box<dyn Iterator<Item = &'a IpAddr> + 'a> {
+        match self {
+            Self::Scalar(inner) => Box::new(std::iter::once(inner)),
+            Self::Vector(inner) => Box::new(inner.iter()),
+        }
+    }
+
+    pub fn into_iter(self) -> impl Iterator<Item = IpAddr> {
+        match self {
+            Self::Scalar(inner) => vec![inner],
+            Self::Vector(inner) => inner,
+        }
+        .into_iter()
+    }
+
+    pub fn first(&self) -> Option<&IpAddr> {
+        self.iter().next()
+    }
+}
+
+#[cfg(test)]
+mod test_cluster_dns_ip {
+    use super::KubernetesClusterDNSIp;
+    use std::net::IpAddr;
+    use std::str::FromStr;
+
+    #[test]
+    fn test_parse_cluster_dns_ip_from_str() {
+        assert_eq!(
+            serde_json::from_str::<KubernetesClusterDNSIp>(r#""127.0.0.1""#).unwrap(),
+            KubernetesClusterDNSIp::Scalar(IpAddr::from_str("127.0.0.1").unwrap())
+        );
+        assert_eq!(
+            serde_json::from_str::<KubernetesClusterDNSIp>(r#""::1""#).unwrap(),
+            KubernetesClusterDNSIp::Scalar(IpAddr::from_str("::1").unwrap())
+        );
+    }
+
+    #[test]
+    fn test_parse_cluster_dns_ip_from_list() {
+        assert_eq!(
+            serde_json::from_str::<KubernetesClusterDNSIp>(r#"[]"#).unwrap(),
+            KubernetesClusterDNSIp::Vector(vec![])
+        );
```

Comment:
If we don't allow an empty string, should we allow an empty list?

I am OK either way as long as this doesn't cause trouble with `kubelet` when rendering the template:
```
[settings.kubernetes]
cluster-dns-ip = []
```

Reply:
The guard in the `kubelet` configs protects the config from an empty list, as empty lists [are considered "falsey" by handlebars](https://github.com/sunng87/handlebars-rust/blob/master/src/json/value.rs#L148).

I've tested empirically and confirmed that an empty list results in `clusterDNS` being absent from the `kubelet` config.

I think an empty string is "less correct" model-wise than an empty list. I'd like the consistency of disallowing both, but I'm not sure it's worth writing the custom deserializer since there's [not an easy way to enforce the constraint in serde](https://github.com/serde-rs/serde/issues/642) otherwise.